src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -527,17 +507,16 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                     {
                         ModifiedCollections = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
                     }
+                    var clusterGuardAddition = options.DisableAtomicDocumentWrites == true
+                        ? null
+                        : GetClusterGuardAddition(command.Index, Database.ClusterTransactionId);
 
                     if (commands != null)
                     {
                         foreach (BlittableJsonReaderObject blittableCommand in commands)
                         {
                             count++;
-                            var changeVector = $"{ChangeVectorParser.RaftTag}:{count}-{Database.DatabaseGroupId}";
-                            if (options.DisableAtomicDocumentWrites == false)
-                            {
-                                changeVector += $",{ChangeVectorParser.TrxnTag}:{command.Index}-{Database.ClusterTransactionId}";
-                            }
+                            var changeVector = GetClusterWideChangeVector(Database.DatabaseGroupId, count, clusterGuardAddition);
```

Comment:
cosmetics, but let's make it
`string GetClusterWideChangeVector(string databaseGroupId, long count, string clusterTransactionId, long index)`
and for legacy
`string GetClusterWideChangeVector(string databaseGroupId, long count)`

Reply:
I like cosmetics but I prefer to avoid reallocate the cluster addition

Reply:
Done